### PR TITLE
Make local development work (in iOS)

### PR DESCRIPTION
**Pull request by honkfestival**
## Overview

This PR does a few things in support of being able to run the entire system locally, especially between the mobile app and server:

 - fixes a typo that prevented the diagnosis keys from being retrieved over the RN bridge
 - updates the exposure configuration URL to match the current server version
 - adds a **Run Exposure Check** item to the test menu when the mock server is disabled
 - adds a `forceRefresh` parameter to the exposure notification service to allow the above item to work
 - updates the `.env` to default to localhost for development

Thank you for pairing with me @henrytao-me! 🙏 

## Testing

Ensure that you have the [server](https://github.com/CovidShield/server) repo checked out and running via e.g. `docker-compose`.

1. Launch the app (I tested in iOS).
1. Open the test menu
1. Disable the mock server
1. Press the "Run Exposure Check" item:
image
1. Observe in the `server` logs that the exposure configuration was retrieved successfully but the diagnosis keys are not because we try to fetch for the current period.

cc @gavrix @jordanAtShopify

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -2,6 +2,6 @@ APP_ID=app.covidshield
 APP_VERSION_CODE=1
 APP_VERSION_NAME=1.0
 TEST_MODE=true
-SUBMIT_URL=https://submit.covidshield.app
-RETRIEVE_URL=https://retrieve.covidshield.app
+SUBMIT_URL=http://localhost:8000
+RETRIEVE_URL=http://localhost:8001
```
Comment (henrytao-me) [inline .env:6]: What I am doing in local development is this:
- Cop `.env` to `.env.local`.
- Change whatever config that I prefer.
- Run native app with:
  + ENVFILE=.env.local yarn run-android.
  + ENVFILE=.env.local yarn run-ios.

So, let's not commit this change.

File: src/screens/home/HomeScreen.tsx [hunk 2]
```diff
@@ -1,27 +1,27 @@
-import React, {useMemo, useState, useEffect} from 'react';
-import {AppState, AppStateStatus} from 'react-native';
-import {BottomSheet, Box} from 'components';
-import {useExposureStatus, useSystemStatus, SystemStatus} from 'services/ExposureNotificationService';
-import {checkNotifications, requestNotifications} from 'react-native-permissions';
-import {useNetInfo} from '@react-native-community/netinfo';
+import React, { useMemo, useState, useEffect } from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+import { BottomSheet, Box } from 'components';
```
Comment (henrytao-me) [inline src/screens/home/HomeScreen.tsx:3]: 👾

Comment (henrytao-me) [reply src/screens/home/HomeScreen.tsx:3]: You might need to download Prettier https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode in VSCode and format your code with `Shift + Option + F`

File: src/services/ExposureNotificationService/ExposureNotificationService.ts [hunk 3]
```diff
@@ -127,13 +127,13 @@ export class ExposureNotificationService {
     return addDays(cycleStart ? new Date(parseInt(cycleStart, 10)) : new Date(), 14);
   }
 
-  async updateExposureStatus(): Promise<ExposureStatus> {
+  async updateExposureStatus(forceRefresh: boolean = false): Promise<ExposureStatus> {
```
Comment (henrytao-me) [inline src/services/ExposureNotificationService/ExposureNotificationService.ts:130]: I think we should have a better way to abstract `forceRefresh`. In general, we don't want any code / logic in Test mode available in production code. I will look into it sometime today.

File: src/testMode/TestMode.tsx [hunk 4]
```diff
@@ -100,6 +100,13 @@ const DrawerContent = () => {
             </>
           )}
         </Section>
+        {!mock.enabled && (
+          <>
+            <Section>
+              <Item title="Run Exposure Check" onPress={() => updateExposureStatus(true)} />
```
Comment (henrytao-me) [inline src/testMode/TestMode.tsx:106]: I would prefer to use `Force Exposure Check`.

**Comment by gavrix** [comment]
I think, conceptually, this might better "Clear exposure check history" ( and perform `this.storage.setItem('lastCheckTimeStamp', '');` in menu action handler) instead of passing the flag down to ExposureNotificationService. The reasoning is to have `ExposureNotificationService` clean from test or debug code.

File: .env (hunk 1 above)
Comment (honkfestival) [reply .env]: OK. Those URLs aren't going to stay valid forever, but until then, I'll at least update them to their correct values.

**Comment by honkfestival** [comment]
@gavrix @henrytao-me 👀 ?

**Pull request merged**